src/Rotations.jl

```diff
@@ -547,19 +553,20 @@ function normal_rotation_distribution(M::Rotations{N}, x, σ::Real) where N
 end
 
 @doc doc"""
-    mean(M::Rotations, x::AbstractVector, w::AbstractWeights; shuffle_rng=nothing, kwargs...)
+    mean(M::Rotations, x::AbstractVector[, w::AbstractWeights]; shuffle_rng=nothing, kwargs...)
 
-Compute the Riemannian mean of `x` using [`GeodesicInterpolationMethod`](@ref).
-If any `x` are not within $\frac{\pi}{2 \sqrt 2}$ of the estimated mean, then
-the estimate is used as the initial guess for the gradient method.
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of `x` using
+[`GeodesicInterpolation`](@ref). If any `x` are not within
+$\frac{\pi}{2 \sqrt 2}$ of the estimated mean, then the estimate is used to
+initialize mean computation using the [`GradientDescent`](@ref).
 """
 mean(::Rotations, args...)
 
-function mean!(M::Rotations, y, x::AbstractVector, w::AbstractWeights; shuffle_rng = nothing, kwargs...)
-    mean!(M, y, x, w, GeodesicInterpolationMethod(); shuffle_rng = shuffle_rng, kwargs...)
+function mean!(M::Rotations, y, x::AbstractVector, w::AbstractVector; shuffle_rng = nothing, kwargs...)
```

Comment:
Why the change back to an abstract vector for the weights?

Reply:
Simply to have fewer type restrictions. `AbstractWeights` is necessary in some cases for dispatch (e.g. for `moment` to distinguish between optional mean `m` and optional weights) and in other places for variance correction (any method calling `var`). Anywhere else, its type is irrelevant, and there's no need to enforce it. Note that the docstrings still use `AbstractWeights`, and it's still the officially supported API.

Reply:
Ok, that sounds reasonable.